lsp-metals.el

```diff
@@ -911,9 +911,11 @@ WORKSPACE is the workspace the client command was received from."
 
 (add-hook 'lsp-after-uninitialized-functions #'lsp-metals--on-workspace-shutdown)
 
+(add-to-list 'lsp-language-id-configuration '(scala-ts-mode . "scala"))
```

Comment:
I'd say this should go into lsp-mode directly.

Reply:
Opened up a PR for this: https://github.com/emacs-lsp/lsp-mode/pull/4106

Reply:
Ah fair enough!

Reply:
Though adding it here as well means you don't have to deal with the case where `scala-ts-mode` is in `:major-modes` for the client defined here, but isn't in `lsp-language-id-configuration` if for example a user has the latest `lsp-metals` version, but an out of date `lsp-mode` version.

`add-to-list` only adds if it isn't already present, so we don't clutter the list keeping it either, wdyt?

Reply:
What will happen in this case? The worst case is that lsp-mode won't work, but it doesn't work for this case a the moment either, since it's a new feature.

Reply:
I found the error:
```
⛔ Warning (lsp-mode): Unable to calculate the languageId for buffer ‘foo.scala’. Take a look at ‘lsp-language-id-configuration’. The ‘major-mode’ is scala-ts-mode
```
Pops up in the warnings buffer.  But yes, it's an edge case.

Reply:
I mean it's a new feature, so it should be expected that it won't work until a user updates packages. It doesn't work now either, so things won't become worse :)

Reply:
Yeah, I think it's unlikely to ever happen anyway, I've removed it in the most recent commit :slightly_smiling_face: